### PR TITLE
Make unit tests run again on AppVeyor

### PR DESCRIPTION
**Pull request by drewnoakes**
Please don't merge this just yet.

### CODE REVIEW

**Comment by drewnoakes** [comment]
I know how to make it build, but it seems that the tests aren't running. NUnit's not quite ready for .NET Core it seems. This may also explain #662. `dotnet test` in its various forms on my local machine also doesn't find/run the NUnit tests. Apparently they don't yet support VS2017's csproj format.

I've used xunit successfully on .NET core and AppVeyor. I'll investigate switching to xunit. Please shout if you see any problem with that.

**Comment by somdoron** [comment]
I actually used custom libraries for NUnit because they didn't release a PR that I made. It is now released so you might want to try to use the official releases. It also what causing the problem, because I didn't update them recently.

**Comment by somdoron** [comment]
Ok I see you already did it :-)

**Comment by drewnoakes** [comment]
Oh I didn't realise that alternative lib was created by you!

I tried a few variations of getting NUnit to work. I'd prefer not having to change test framework, but it seems the NUnit stuff lags behind xUnit when it comes to support and community.

So far conversion to xUnit seems to be ok. By default xUnit runs tests in parallel, which doesn't seem to work with the global context.

**Comment by somdoron** [comment]
I'm fine with changing the framework

**Comment by drewnoakes** [comment]
I'll push the xUnit stuff to this PR for review shortly. It seems to work well, although occasionally some of the proxy tests throw `AddressAlreadyInUseException`, so there's something left to fix.

```text
NetMQ.AddressAlreadyInUseException : Cannot bind address ( inproc://frontend ) - already in use.
   at NetMQ.Core.SocketBase.Bind(String addr) in C:\Users\Drew\Dev\netmq\src\NetMQ\Core\SocketBase.cs:line 435
   at NetMQ.NetMQSocket.Bind(String address) in C:\Users\Drew\Dev\netmq\src\NetMQ\NetMQSocket.cs:line 183
   at NetMQ.Tests.ProxyTests.ControlSocketObservedMessages() in C:\Users\Drew\Dev\netmq\src\NetMQ.Tests\ProxyTests.cs:line 46
```

I'd appreciate some help working out why the proxy tests can fail.

**Comment by drewnoakes** [comment]
Pushed. These tests pass on my machine, most of the time (proxy tests, as discussed).

File: src/NetMQ.Tests/ProxyTests.cs [hunk 1]
```diff
@@ -1,29 +1,13 @@
...
+using Xunit;
 
 namespace NetMQ.Tests
 {
-    [TestFixture]
-    public class ProxyTests
+    public class ProxyTests : IClassFixture<NetMQConfigFixture>
     {
-        [OneTimeSetUp]
-        public void SetUp()
```
Comment (somdoron) [inline src/NetMQ.Tests/ProxyTests.cs]: I think this is the reason the tests fail.
Does xUnit has an alternative for this?

**Comment by somdoron** [comment]
Try to call NetMQConfig.Cleanup before the proxy tests.

File: src/NetMQ.Tests/ProxyTests.cs (hunk 1 above)
Comment (drewnoakes) [inline src/NetMQ.Tests/ProxyTests.cs]: The `IClassFixture<NetMQConfigFixture>` interface causes xUnit to run `NetMQConfigFixture.ctor` first, then run all the tests, then run `NetMQConfigFixture.Dispose`. So `Cleanup(true)` runs at the both the start and end of the fixture.

I'll try running it at the start and end of every test instead.

**Comment by somdoron** [comment]
Let me know when to merge

**Comment by drewnoakes** [comment]
I tried several combinations of settings for Travis to get a build working, and none of them succeeded. There are many pitfalls here. It seems that building a 2017 csproj that targets full .NET Framework on a Linux build server is not yet straightforward.

When I did find a combination of settings that got tests running in at least one configuration, they would hang and time out after ten minutes.

Despite this, I think this PR can now be merged. What it does do is get the unit tests working on AppVeyor. Currently the tests aren't running anywhere, so this is an improvement. If merged it should fix #662 and #575.

**Comment by drewnoakes** [comment]
I think this PR is good to merge. It doesn't actually fix Travis CI, but it does make the tests run again on AppVeyor, as well as fix a few bugs that the tests would have picked up if they were running.

**Pull request merged**